### ISSUE TITLE
filtering a complex string

### ISSUE THREAD

**Issue by CharlesNepote** [bi-tm/express-nedb-rest#10]
Hi Torsten,
I'm trying to query (with my browser):
```http://127.0.0.1:8100/test?$single&$filter=uri%20$eq%20https://gist.githubusercontent.com/CharlesNepote/26126a30adddc1781cb652b2af164ea8/raw/92ea63558c3d615ac21c5c31b286fa5f60cf7cd0/prenom-schema.json```

the result is:
```
unvalid $filter Lexical error on line 1. Unrecognized text.
uri $eq https://gist.githubuserco
-------------^
```

I guess querying with an URL is for simple thing. "put the filter as JSON string into HTTP body" as you suggested earlier, might be a good idea -- would it also work with a browser?
Regards, Charles.

**Comment by bi-tm** [comment]
Hello Charles,

i modified the $filter parser in version 1.2.3.
My test tool encodes the query string (as the browser does), so there should work:
`/rest/test?$single&$filter=uri $eq https://gist.githubusercontent.com/CharlesNepote/26126a30adddc1781cb652b2af164ea8/raw/92ea63558c3d615ac21c5c31b286fa5f60cf7cd0/prenom-schema.json`

But if you make a `XMLHttpRequest` in JavaScript, please use `encodeUri` before you send the request:
`/rest/test?$single&$filter=uri%20$eq%20https://gist.githubusercontent.com/CharlesNepote/26126a30adddc1781cb652b2af164ea8/raw/92ea63558c3d615ac21c5c31b286fa5f60cf7cd0/prenom-schema.json"`

Regards
Torsten

**Comment by bi-tm** [comment]
Hello Charles,
i tried to implement the filter in HTTP body, as i suggested. Unfortunately expressJS doesn't parse the body during a GET call. So this is not the right way. Do you have another suggestion for complex filters?
Regards
Torsten

**Comment by CharlesNepote** [comment]
Hi Torsten,

Your solution implemented in version 1.2.3 is working and, for the moment, I didn't experiment cases where I would need a better functionality. Thanks a lot!

I finally published a **very early** version of my app: https://github.com/CharlesNepote/csv-schema-playground

I'm not a programmer nor an engineer so it might be plenty of bad things. You will be able to see how I used `express-nedb-rest` and better understand what I want to do.

Best regards, Charles.
